streams/src/main/java/org/apache/kafka/streams/state/internals/Segments.java

```diff
@@ -66,7 +67,7 @@ long segmentId(long timestamp) {
     }
 
     String segmentName(long segmentId) {
-        return name + "-" + formatter.format(new Date(segmentId * segmentInterval));
+        return name + ":" + segmentId;
```

Comment:
Should we add a comment about the old format to avoid the usage of `-` for another potential format change in the future?